zmoney/Services/CacheService/Extensions/Transaction+Storable.swift

```diff
@@ -92,8 +96,18 @@ extension Transaction: Storable {
         self.incomeBankID = entity.incomeBankID
         self.outcomeBankID = entity.outcomeBankID
         self.reminderMarker = entity.reminderMaker
-        self.categories = entity.categories
-        self.incomeTransactionInstrument = entity.incomeTransactionInstrument
-        self.outcomeTransactionInstrument = entity.outcomeTransactionInstrument
+        if let categories = entity.categories {
+            for category in categories {
+                if let categoryTag = category as? Tag {
+                    self.categories?.append(categoryTag)
+                }
+            }
+        }
```

Comment:
А вот здесь можно что-то типа такого сделать

```suggestion
        let tags = entity.categories?.compactMap({ $0 as? Tag }) ?? []
        categories?.append(contentsOf: tags)
```